### PR TITLE
1. Created separate classes for address information(block, street, un…

### PR DESCRIPTION
**Pull request by AngularJiaSheng**
…it, postalCode).

2. 1 retrival method is created doe each class, however, still not accessible for users.
3. The class is done by spliting address string using the split method from String API

### CODE REVIEW

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -48,6 +48,7 @@ public AddressBook(UniquePersonList persons, UniqueTagList tags) {
         }
     }
 
+
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/AddressBook.java]: Extra line

File: src/seedu/addressbook/data/AddressBook.java [hunk 2]
```diff
@@ -13,6 +8,11 @@
 import seedu.addressbook.data.tag.Tag;
 import seedu.addressbook.data.tag.UniqueTagList;
 
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/AddressBook.java]: Avoid formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes. If you wish to commit formatting changes to unrelated parts of the code, do so in a separate commit.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -14,19 +14,28 @@
...
     private boolean isPrivate;
+    private String[] addressStringSplitUp;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Extra line

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -14,19 +14,28 @@
...
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        addressStringSplitUp = trimmedAddress.split(", ");
+        Block block = new Block(addressStringSplitUp[0]);
+        Street street = new Street(addressStringSplitUp[1]);
+        Unit unit = new Unit(addressStringSplitUp[2]);
+        PostalCode postalCode = new PostalCode(addressStringSplitUp[3]);
+        //Test calling getBlock() from Block class.
+        // System.out.println(block.getBlock());
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -14,19 +14,28 @@
...
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        addressStringSplitUp = trimmedAddress.split(", ");
+        Block block = new Block(addressStringSplitUp[0]);
+        Street street = new Street(addressStringSplitUp[1]);
+        Unit unit = new Unit(addressStringSplitUp[2]);
+        PostalCode postalCode = new PostalCode(addressStringSplitUp[3]);
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: What if the address provided has less than 3 commas?

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
         return isPrivate;
     }
 }
+
+
+class Block{
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Is there a reason for having no explicit access modifier?

File: src/seedu/addressbook/data/person/Address.java [hunk 7]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
...
+    String block;
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: - Is there a reason for having no explicit access modifier? 
- Avoid naming your variables like classes.

File: src/seedu/addressbook/data/person/Address.java [hunk 8]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
...
+    public Block(String block){
+
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Extra line.

File: src/seedu/addressbook/data/person/Address.java [hunk 9]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
...
+        this.block = block ;
+    }
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Missing new line in between blocks

File: src/seedu/addressbook/data/person/Address.java [hunk 10]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
...
+    public String getBlock(){
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Missing space before opening brace.

File: src/seedu/addressbook/data/person/Address.java [hunk 11]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
...
+        return this.block;
+    }
+
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Extra line.

File: src/seedu/addressbook/data/person/Address.java [hunk 12]
```diff
@@ -57,3 +66,47 @@ public boolean isPrivate() {
         return isPrivate;
     }
 }
+
+
+class Block{
+    String block;
+    public Block(String block){
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Missing space before opening brace.

File: src/seedu/addressbook/data/person/Address.java [hunk 13]
```diff
@@ -14,19 +14,28 @@
 
     public final String value;
     private boolean isPrivate;
+    private String[] addressStringSplitUp;
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Address.java]: Does this need to be an instance variable? Choose the narrowest possible scope for variables. If it can be a local variable inside a method, don't declare it as an instance variable.

**Comment by PierceAndy** [comment]
- Remember to update tests and documentations in the future.
- Your commit messages can be improved. Refer to https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.adoc#commit-message
- There are style guide issues. Refer to https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc
(These links can be found in `README.md` of `addressbook-level2`: https://github.com/nus-cs2103-AY1718S1/addressbook-level2#contact-us)

@AngularJiaSheng Some comments added. Please close the PR after reading comments.

**Comment by AngularJiaSheng** [comment]
Okay! Thank you so much for your feedback!
